sonarts-core/src/utils/sonarAnalysis.ts

```diff
@@ -132,8 +132,16 @@ export class SonarIssue extends tslint.RuleFailure {
     return this;
   }
 
-  public addSecondaryLocation(secondaryLocation: IssueLocation): SonarIssue {
-    this.secondaryLocations.push(secondaryLocation);
+  public addSecondaryLocation(node: ts.Node, message: string): SonarIssue;
+  public addSecondaryLocation(secondaryLocation: IssueLocation): SonarIssue;
+  public addSecondaryLocation(): SonarIssue {
+    if (arguments.length == 1) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: == should be === ([typescript:S1440](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1440))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=issue-476&issues=AWGdtLPMR7xclXvFe6qP&open=AWGdtLPMR7xclXvFe6qP)

[ISSUE_KEY]: AWGdtLPMR7xclXvFe6qP